### ISSUE TITLE
Errors when launching in Kubuntu 11.04 after install via bootstrap.sh

### ISSUE THREAD

**Issue by Hoverbear** [spf13/spf13-vim#25]
Output of installer, then attempting to launch vim:

hoverbear@turing:~$ curl https://raw.github.com/spf13/spf13-vim/master/bootstrap.sh -o - | sh
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   674  100   674    0     0     63      0  0:00:10  0:00:10 --:--:--  1685
thanks for installing spf13-vim

backing up current vim config

cloning spf13-vim

Cloning into /home/hoverbear/.spf13-vim...
remote: Counting objects: 1217, done.
remote: Compressing objects: 100% (652/652), done.
remote: Total 1217 (delta 548), reused 1162 (delta 506)
Receiving objects: 100% (1217/1217), 934.91 KiB | 613 KiB/s, done.
Resolving deltas: 100% (548/548), done.
Submodule '.vim/bundle/VimDebugger' (git://github.com/taxilian/VimDebugger.git) registered for path '.vim/bundle/VimDebugger'
Submodule '.vim/bundle/ack' (git://github.com/mileszs/ack.vim.git) registered for path '.vim/bundle/ack'
Submodule '.vim/bundle/checksyntax' (http://github.com/tomtom/checksyntax_vim.git) registered for path '.vim/bundle/checksyntax'
Submodule '.vim/bundle/command-t' (git://github.com/wincent/Command-T.git) registered for path '.vim/bundle/command-t'
Submodule '.vim/bundle/delimitMate' (http://github.com/Raimondi/delimitMate.git) registered for path '.vim/bundle/delimitMate'
Submodule '.vim/bundle/easytags' (http://github.com/spf13/vim-easytags.git) registered for path '.vim/bundle/easytags'
Submodule '.vim/bundle/fugitive' (http://github.com/tpope/vim-fugitive.git) registered for path '.vim/bundle/fugitive'
Submodule '.vim/bundle/jslint' (git://github.com/bloveridge/jslint.vim.git) registered for path '.vim/bundle/jslint'
Submodule '.vim/bundle/nerdcommenter' (git://github.com/scrooloose/nerdcommenter.git) registered for path '.vim/bundle/nerdcommenter'
Submodule '.vim/bundle/nerdtree' (git://github.com/spf13/nerdtree.git) registered for path '.vim/bundle/nerdtree'
Submodule '.vim/bundle/piv' (git://github.com/spf13/PIV.git) registered for path '.vim/bundle/piv'
Submodule '.vim/bundle/pydoc.vim' (https://github.com/fs111/pydoc.vim.git) registered for path '.vim/bundle/pydoc.vim'
Submodule '.vim/bundle/pyflakes' (git://github.com/bloveridge/pyflakes-vim.git) registered for path '.vim/bundle/pyflakes'
Submodule '.vim/bundle/snipmate' (git://github.com/spf13/snipmate.vim.git) registered for path '.vim/bundle/snipmate'
Submodule '.vim/bundle/snipmate-snippets/snippets' (git://github.com/spf13/snipmate-snippets.git) registered for path '.vim/bundle/snipmate-snippets/snippets'
Submodule '.vim/bundle/supertab' (git://github.com/ervandew/supertab.git) registered for path '.vim/bundle/supertab'
Submodule '.vim/bundle/surround' (git://github.com/spf13/vim-surround.git) registered for path '.vim/bundle/surround'
Submodule '.vim/bundle/vim-colors-solarized' (https://github.com/altercation/vim-colors-solarized.git) registered for path '.vim/bundle/vim-colors-solarized'
Submodule '.vim/bundle/vim-json' (git://github.com/leshill/vim-json.git) registered for path '.vim/bundle/vim-json'
Submodule '.vim/bundle/vim-markdown' (git://github.com/spf13/vim-markdown.git) registered for path '.vim/bundle/vim-markdown'
Cloning into .vim/bundle/VimDebugger...
remote: Counting objects: 80, done.
remote: Compressing objects: 100% (77/77), done.
remote: Total 80 (delta 36), reused 8 (delta 2)
Receiving objects: 100% (80/80), 46.04 KiB, done.
Resolving deltas: 100% (36/36), done.
Submodule path '.vim/bundle/VimDebugger': checked out 'f59e0432ebde5cdd120c6aab942208a1cfb3b16a'
Cloning into .vim/bundle/ack...
remote: Counting objects: 175, done.
remote: Compressing objects: 100% (98/98), done.
remote: Total 175 (delta 73), reused 144 (delta 45)
Receiving objects: 100% (175/175), 19.96 KiB, done.
Resolving deltas: 100% (73/73), done.
Submodule path '.vim/bundle/ack': checked out 'a41d5d52c39a31128e969e69acf800b198cb07f9'
Cloning into .vim/bundle/checksyntax...
remote: Counting objects: 152, done.
remote: Compressing objects: 100% (67/67), done.
remote: Total 152 (delta 62), reused 145 (delta 56)
Receiving objects: 100% (152/152), 24.55 KiB, done.
Resolving deltas: 100% (62/62), done.
Submodule path '.vim/bundle/checksyntax': checked out 'a6ddb09497dd8b3da07da8bb4ceb9e7c1c286bba'
Cloning into .vim/bundle/command-t...
remote: Counting objects: 2267, done.
remote: Compressing objects: 100% (945/945), done.
remote: Total 2267 (delta 1227), reused 2160 (delta 1122)
Receiving objects: 100% (2267/2267), 368.91 KiB | 391 KiB/s, done.
Resolving deltas: 100% (1227/1227), done.
Submodule path '.vim/bundle/command-t': checked out 'cee619ac6e906d1c6bfc3052f01ccaa6ac1d4b3c'
Submodule 'vendor/vimball' (git://github.com/tomtom/vimball.rb.git) registered for path 'vendor/vimball'
Submodule 'vendor/vimscriptuploader' (git://github.com/tomtom/vimscriptuploader.rb.git) registered for path 'vendor/vimscriptuploader'
Cloning into vendor/vimball...
remote: Counting objects: 26, done.
remote: Compressing objects: 100% (26/26), done.
remote: Total 26 (delta 8), reused 0 (delta 0)
Receiving objects: 100% (26/26), 10.56 KiB, done.
Resolving deltas: 100% (8/8), done.
Submodule path 'vendor/vimball': checked out 'f60b56b1ed6eac51fbf78549a4b482b43aa45e6a'
Cloning into vendor/vimscriptuploader...
remote: Counting objects: 20, done.
remote: Compressing objects: 100% (20/20), done.
remote: Total 20 (delta 6), reused 0 (delta 0)
Receiving objects: 100% (20/20), 7.59 KiB, done.
Resolving deltas: 100% (6/6), done.
Submodule path 'vendor/vimscriptuploader': checked out 'dade97a78899ba2836baf6897b4c60594a17f641'
Cloning into .vim/bundle/delimitMate...
remote: Counting objects: 948, done.
remote: Compressing objects: 100% (402/402), done.
remote: Total 948 (delta 445), reused 846 (delta 393)
Receiving objects: 100% (948/948), 188.35 KiB | 349 KiB/s, done.
Resolving deltas: 100% (445/445), done.
Submodule path '.vim/bundle/delimitMate': checked out '9efb01da683721927589e7e340e8f2374d7dd3ea'
Cloning into .vim/bundle/easytags...
remote: Counting objects: 237, done.
remote: Compressing objects: 100% (109/109), done.
remote: Total 237 (delta 125), reused 233 (delta 122)
Receiving objects: 100% (237/237), 60.59 KiB, done.
Resolving deltas: 100% (125/125), done.
Submodule path '.vim/bundle/easytags': checked out '6db62ef85cdf82451a1e6137ed5edd8e38935b22'
Cloning into .vim/bundle/fugitive...
remote: Counting objects: 731, done.
remote: Compressing objects: 100% (289/289), done.
remote: Total 731 (delta 280), reused 701 (delta 253)
Receiving objects: 100% (731/731), 102.67 KiB, done.
Resolving deltas: 100% (280/280), done.
Submodule path '.vim/bundle/fugitive': checked out '44533707686c09972bf845c31544368bdfe0ce4f'
Cloning into .vim/bundle/jslint...
remote: Counting objects: 335, done.
remote: Compressing objects: 100% (146/146), done.
remote: Total 335 (delta 169), reused 304 (delta 145)
Receiving objects: 100% (335/335), 88.37 KiB, done.
Resolving deltas: 100% (169/169), done.
Submodule path '.vim/bundle/jslint': checked out '72d34cc3f1d6f0b95ad935edfc38b40c9be0d88c'
Cloning into .vim/bundle/nerdcommenter...
remote: Counting objects: 979, done.
remote: Compressing objects: 100% (374/374), done.
remote: Total 979 (delta 357), reused 932 (delta 317)
Receiving objects: 100% (979/979), 246.51 KiB | 99 KiB/s, done.
Resolving deltas: 100% (357/357), done.
Submodule path '.vim/bundle/nerdcommenter': checked out '5a9371437f97eef12325c04607de5e58b7168432'
Cloning into .vim/bundle/nerdtree...
remote: Counting objects: 1987, done.
remote: Compressing objects: 100% (639/639), done.
remote: Total 1987 (delta 880), reused 1953 (delta 851)
Receiving objects: 100% (1987/1987), 896.11 KiB | 206 KiB/s, done.
Resolving deltas: 100% (880/880), done.
Submodule path '.vim/bundle/nerdtree': checked out 'c2e3bbe762bd1a64f7f949cf07acdd4e480d1519'
Cloning into .vim/bundle/piv...
remote: Counting objects: 7892, done.
remote: Compressing objects: 100% (4983/4983), done.
remote: Total 7892 (delta 2898), reused 7875 (delta 2891)
Receiving objects: 100% (7892/7892), 3.46 MiB | 1.33 MiB/s, done.
Resolving deltas: 100% (2898/2898), done.
Submodule path '.vim/bundle/piv': checked out '25deca4e2465b09812a374ba26e81b1f147706f0'
Cloning into .vim/bundle/pydoc.vim...
remote: Counting objects: 56, done.
remote: Compressing objects: 100% (48/48), done.
remote: Total 56 (delta 22), reused 0 (delta 0)
Unpacking objects: 100% (56/56), done.
Submodule path '.vim/bundle/pydoc.vim': checked out 'fd814c429e5269da10df87429b62c570617456f2'
Cloning into .vim/bundle/pyflakes...
remote: Counting objects: 118, done.
remote: Compressing objects: 100% (61/61), done.
remote: Total 118 (delta 57), reused 110 (delta 54)
Receiving objects: 100% (118/118), 16.96 KiB, done.
Resolving deltas: 100% (57/57), done.
Submodule path '.vim/bundle/pyflakes': checked out '8140709d8cef70dd441c0a0fe25db2fa218cd010'
Submodule 'ftplugin/python/pyflakes' (git://github.com/kevinw/pyflakes.git) registered for path 'ftplugin/python/pyflakes'
Cloning into ftplugin/python/pyflakes...
remote: Counting objects: 195, done.
remote: Compressing objects: 100% (95/95), done.
remote: Total 195 (delta 111), reused 170 (delta 92)
Receiving objects: 100% (195/195), 58.70 KiB, done.
Resolving deltas: 100% (111/111), done.
Submodule path 'ftplugin/python/pyflakes': checked out 'd7837f67e778659bd00428a1a84160dbbaf40438'
Cloning into .vim/bundle/snipmate...
remote: Counting objects: 1381, done.
remote: Compressing objects: 100% (557/557), done.
remote: Total 1381 (delta 733), reused 1271 (delta 655)
Receiving objects: 100% (1381/1381), 223.65 KiB | 115 KiB/s, done.
Resolving deltas: 100% (733/733), done.
Submodule path '.vim/bundle/snipmate': checked out '14b6970ac641a91ea22d3d52461309e70a9259a9'
Cloning into .vim/bundle/snipmate-snippets/snippets...
remote: Counting objects: 3305, done.
remote: Compressing objects: 100% (1477/1477), done.
remote: Total 3305 (delta 646), reused 3166 (delta 559)
Receiving objects: 100% (3305/3305), 319.00 KiB | 129 KiB/s, done.
Resolving deltas: 100% (646/646), done.
Submodule path '.vim/bundle/snipmate-snippets/snippets': checked out 'a9300419ba8b735c51e2bbd58a2e87aff5c4bc9b'
Cloning into .vim/bundle/supertab...
remote: Counting objects: 276, done.
remote: Compressing objects: 100% (93/93), done.
remote: Total 276 (delta 99), reused 269 (delta 93)
Receiving objects: 100% (276/276), 50.39 KiB, done.
Resolving deltas: 100% (99/99), done.
Submodule path '.vim/bundle/supertab': checked out '80ec6539e4139a2e0281a0f1e36d5038d55ad980'
Cloning into .vim/bundle/surround...
remote: Counting objects: 203, done.
remote: Compressing objects: 100% (83/83), done.
remote: Total 203 (delta 50), reused 201 (delta 50)
Receiving objects: 100% (203/203), 44.22 KiB, done.
Resolving deltas: 100% (50/50), done.
Submodule path '.vim/bundle/surround': checked out '27710a2224d6dd0486d1c40d09ef18dd752e7d37'
Cloning into .vim/bundle/vim-colors-solarized...
remote: Counting objects: 336, done.
remote: Compressing objects: 100% (189/189), done.
remote: Total 336 (delta 102), reused 308 (delta 75)
Receiving objects: 100% (336/336), 91.70 KiB, done.
Resolving deltas: 100% (102/102), done.
Submodule path '.vim/bundle/vim-colors-solarized': checked out '528a59f26d12278698bb946f8fb82a63711eec21'
Cloning into .vim/bundle/vim-json...
remote: Counting objects: 6, done.
remote: Compressing objects: 100% (3/3), done.
remote: Total 6 (delta 0), reused 0 (delta 0)
Receiving objects: 100% (6/6), done.
Submodule path '.vim/bundle/vim-json': checked out '0bad66cc6a832959c69eb81459ae8c993f04bb54'
Cloning into .vim/bundle/vim-markdown...
remote: Counting objects: 72, done.
remote: Compressing objects: 100% (30/30), done.
remote: Total 72 (delta 23), reused 70 (delta 23)
Receiving objects: 100% (72/72), 7.56 KiB, done.
Resolving deltas: 100% (23/23), done.
Submodule path '.vim/bundle/vim-markdown': checked out '948c411ba5eca7d4ce113e6574b97d5f50b92ed2'
building command-t executable

command-t depends on ruby and rake to be present

(in /home/hoverbear/.spf13-vim/.vim/bundle/command-t)
/usr/bin/ruby1.8 extconf.rb
checking for ruby.h... yes
creating Makefile
gcc -I. -I/usr/lib/ruby/1.8/x86_64-linux -I/usr/lib/ruby/1.8/x86_64-linux -I. -DHAVE_RUBY_H    -fPIC -fno-strict-aliasing -g -g -O2  -fPIC   -std=c99 -Wall -Wextra -Wno-unused-parameter -c ext.c
gcc -I. -I/usr/lib/ruby/1.8/x86_64-linux -I/usr/lib/ruby/1.8/x86_64-linux -I. -DHAVE_RUBY_H    -fPIC -fno-strict-aliasing -g -g -O2  -fPIC   -std=c99 -Wall -Wextra -Wno-unused-parameter -c matcher.c
gcc -I. -I/usr/lib/ruby/1.8/x86_64-linux -I/usr/lib/ruby/1.8/x86_64-linux -I. -DHAVE_RUBY_H    -fPIC -fno-strict-aliasing -g -g -O2  -fPIC   -std=c99 -Wall -Wextra -Wno-unused-parameter -c match.c
gcc -shared -o ext.so ext.o matcher.o match.o -L. -L/usr/lib -L. -Wl,-Bsymbolic-functions -rdynamic -Wl,-export-dynamic    -lruby1.8  -lpthread -lrt -ldl -lcrypt -lm   -lc

hoverbear@turing:~$ vim
Error detected while processing function <SNR>13_VimDebuggerLoad:
line    2:  
E117: Unknown function: <SNR>13_VimDebuggerInit  
Error detected while processing function DefPython:  
line  550:  
Traceback (most recent call last):  
  File "<string>", line 547, in <module>  
  File "<string>", line 397, in **init**  
NameError: global name 'VimDebugger' is not defined  
Error detected while processing /home/hoverbear/.spf13-vim/.vim/bundle/snipmate/plugin/snipMate.vim:  
line   31:  
E117: Unknown function: funcref#Function  
E116: Invalid arguments for function get(s:snipMate, 'get_snippets', funcref#Function("snipMate#GetSnippets"))  
E15: Invalid expression: get(s:snipMate, 'get_snippets', funcref#Function("snipMate#GetSnippets"))  
Press ENTER or type command to continue

**Comment by spf13** [comment]
Try gvim or Macvim. Your vim isn't compiled with the right stuff. It usually
degrades gracefully.  What os/ distro are you running?

**Comment by Hoverbear** [comment]
Kubuntu 11.04.

**Comment by elliottneilclark** [comment]
I have this as well on MacVim build 57
And on ubuntu 10.10 with vim-gtk

Both install echo 1 for has ruby and had python.

**Comment by lasfox** [comment]
I got the same issue on ubuntu 11.04.
Just have a bit diff .

anson$ vim
处理 function <SNR>14_VimDebuggerLoad 时发生错误:
第    2 行:
E117: 未定义的函数: <SNR>14_VimDebuggerInit
处理 function DefPython 时发生错误:
第  550 行:
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in **init**
NameError: global name 'VimDebugger' is not defined

**Comment by elliottneilclark** [comment]
```
Error detected while processing /home/eclark/.spf13-vim/.vim/bundle/snipmate/plugin/snipMate.vim:
line   31:
E117: Unknown function: funcref#Function
E116: Invalid arguments for function get(s:snipMate, 'get_snippets', funcref#Function("snipMate#GetSnippets"))
E15: Invalid expression: get(s:snipMate, 'get_snippets', funcref#Function("snipMate#GetSnippets"))
Error detected while processing function <SNR>35_VimDebuggerLoad:
line    2:
E117: Unknown function: <SNR>35_VimDebuggerInit
Error detected while processing function DefPython:
line  550:
Traceback (most recent call last):
  File "<string>", line 547, in <module>
  File "<string>", line 397, in __init__
NameError: global name 'VimDebugger' is not defined
Press ENTER or type command to continue
```

**Comment by ngaloppo** [comment]
I have the same issue with gvim:

Error detected while processing C:\Users\nico.vim\bundle\snipmate\plugi
n\snipMate.vim:

line   31:

E117: Unknown function: funcref#Function

E116: Invalid arguments for function get(s:snipMate, 'get_snippets', 
funcref#Function("snipMate#GetSnippets"))

E15: Invalid expression: get(s:snipMate, 'get_snippets', funcref#Function("sn
ipMate#GetSnippets"))

**Comment by spf13** [comment]
Thanks everyone for the feedback. I'll need to install ubuntu in a vm to test this out. In the meantime, could someone test out the 3.0 branch of spf13-vim and see if the issue exists there as well?

**Comment by elliottneilclark** [comment]
For me the 3.0 branch fixes the issues on mac an ubuntu

**Comment by spf13** [comment]
This is fixed in the 3.0 branch (which is now the default). Please upgrade to that.
